src/components/enter-password/enter-password-controller.ts

```diff
@@ -29,8 +32,36 @@ const ENTER_PASSWORD_ACCOUNT_EXISTS_TEMPLATE =
 const ENTER_PASSWORD_ACCOUNT_EXISTS_VALIDATION_KEY =
   "pages.enterPasswordAccountExists.password.validationError.incorrectPassword";
 
-export function enterPasswordGet(req: Request, res: Response): void {
-  res.render(ENTER_PASSWORD_TEMPLATE);
+export function enterPasswordGet(
+  service: CheckReauthServiceInterface = checkReauthUsersService()
+): ExpressRouteFunc {
+  return async function (req: Request, res: Response) {
+    if (!supportReauthentication()) {
```

Comment:
I'm wondering if for readability we also pull the `isReauthenticationRequired` condition up to this bit. So the condition here is if either we don't support reauth, or reauth is not required, then early return and none of the rest of this code is relevant to the code path